### ISSUE TITLE
[Wave 6: Workspace Chats] GBR is not shown in workspace chat which has a request waiting on payment

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#32779]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.10-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1702069357077369

## Action Performed:
1. As a member request money from the workspace chat
2. Login as a admin

## Expected Result:
LHN should show chat with a GBR as it needs a payment

## Actual Result:
WS chat does not have a GBR and it shows up only after opening the report

(I’m guessing the pusher request that sent the new open report, cleared the hasOutstandingChildRequest field.)

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/fdf07d09-816f-466a-8cd5-2c38f5ef8a00

![image (3)](https://github.com/Expensify/App/assets/38435837/8ea92d7d-308c-4c6f-8b32-1bcd8e346927)

![image (2)](https://github.com/Expensify/App/assets/38435837/877d1fa8-a0a1-4616-a968-e769dd065850)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011e391b7a0331eaaf</li>
        <li>Upwork Job ID: 1734239151890964480</li>
        <li>Last Price Increase: 2023-12-11</li>
    </ul>
</details>

**Comment by yuwenmemon** [comment]
There might be more at play here...

Looking at the api requests, it looks like we call OpenReport with the correct data:
![Screenshot 2023-12-12 at 8 29 23 PM](https://github.com/Expensify/App/assets/4741899/3b0f40be-0428-434f-81f4-917b3697ca8e)

But then later in OpenApp, we get the wrong data:
![Screenshot 2023-12-12 at 8 29 33 PM](https://github.com/Expensify/App/assets/4741899/d9a839d5-4759-447c-9cc6-2968742d1017)

cc @mountiny

**Comment by yuwenmemon** [comment]
Yep, confirming my theory is that the GBR shows when you call `OpenReport` again by clicking in and out of the report:

https://github.com/Expensify/App/assets/4741899/908de389-9cc9-4b17-b476-cb6833bba66b

**Comment by yuwenmemon** [comment]
Ah, I figured it out. It's because [this line](https://github.com/Expensify/Auth/blob/d633f5008ce3b6abd17014ecdaefcfe6537b14e8/auth/lib/Report.cpp#L5521) is in a for-loop so the `reports[parentReportIndexIterator->second]["hasOutstandingChildRequest"]` entry can get overwritten if another one returns `false`

**Comment by mountiny** [comment]
Commented on the PR

**Comment by yuwenmemon** [comment]
> Should https://github.com/Expensify/App/issues/32924 be handled here?

I think that's a separate issue actually.

**Comment by yuwenmemon** [comment]
The PR fixing this has been merged and we should be good to close this out once it's deployed.

**Comment by mountiny** [comment]
I think we are good to close this now
